source/NVDAObjects/UIA/__init__.py

```diff
@@ -106,21 +172,37 @@ def _get_NVDAObjectAtStart(self):
 	def _get_bookmark(self):
 		return self.copy()
 
-	def _getControlFieldForObject(self, obj):
+	def _getControlFieldForObject(self, obj,isEmbedded=False,startOfNode=False,endOfNode=False):
+		"""
+		Fetch control field information for the given UIA NVDAObject.
+		@ param obj: the NVDAObject the control field is for.
+		@type obj: L{UIA}
+		@param isEmbedded: True if this NVDAObject is for a leaf node (has no useful children).
+		@ type isEmbedded: bool
+		@param startOfNode: True if the control field represents the very start of this object.
+		@type startOfNode: bool
+		@param endOfNode: True if the control field represents the very end of this object.
+		@type endOfNode: bool
+		@return: The control field for this object
+		@rtype: dict containing NVDA control field data.
+		"""
 		role = obj.role
 		field = textInfos.ControlField()
+		# Ensure this controlField is unique to the object
+		field['runtimeID']=obj.UIAElement.getRuntimeID()
+		field['_startOfNode']=startOfNode
+		field['_endOfNode']=endOfNode
 		field["role"] = obj.role
 		states = obj.states
 		# The user doesn't care about certain states, as they are obvious.
 		states.discard(controlTypes.STATE_EDITABLE)
 		states.discard(controlTypes.STATE_MULTILINE)
 		states.discard(controlTypes.STATE_FOCUSED)
 		field["states"] = states
-		# Only include name if the object's inner text is empty.
-		# could be unperformant.
 		text=self.obj.makeTextInfo(obj).text
 		if not text or text.isspace():
-			field["name"] = obj.name
+			field['name']=obj.name

```

Comment:
So name only gets exposed if there's no text. How does this not break quick navigation to edit fields containing text? I just tested and it doesn't, but I don't quite understand why.

I'm not sure if this is related, but I notice the text of an edit field doesn't get read when quick navigating. For example:

`<input type="text" aria-label="cow" value="goat">`

If I press e to move to that field, it says "cow edit", but you never hear "goat".